core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -557,17 +555,31 @@ object CachedDataFrame
     m
   }
 
+  // Some helper fields to show correct plan Info on UI as
+  // per current set of constants even though a cached plan
+  // is executed
+  var pls: Map[Int, ParamLiteral] = null
+  val emptyParameter = ParamLiteral(-1, IntegerType, -1)
+
   private[sql] def queryPlanInfo(plan: SparkPlan,
-      allLiterals: Array[LiteralValue]): SparkPlanInfo = PartitionedPhysicalScan.getSparkPlanInfo(
-    plan.transformAllExpressions {
-      case ParamLiteral(_v, _dt, _p) =>
-        val x = allLiterals.find(_.position == _p)
-        val v = x match {
-          case Some(LiteralValue(_, _, _)) => x.get.value
-          case None => _v
+      allLiterals: Array[LiteralValue]): SparkPlanInfo = {
+    if (allLiterals.nonEmpty) {
+      if (pls == null) {
+        pls = Map.empty
+        plan.transformAllExpressions {
+          case pl@ParamLiteral(_, _, _p) =>
+            pls = pls + (_p -> pl)
+            pl
         }
-        Literal(v, _dt)
-    })
+      }
+      if (allLiterals.length == pls.size) {
+        allLiterals.map(x => {
```

Comment:
change to foreach instead of map else it will unnecessarily create a Seq[Unit]

Reply:
Ok.